src/libtls/tests/suites/test_socket.c

```diff
@@ -493,6 +493,98 @@ static void test_tls_curves(tls_version_t version, uint16_t port, bool cauth,
 	free(config);
 }
 
+/**
+ * TLS server version test wrapper function
+ */
+static void test_tls_server(tls_version_t version, uint16_t port, bool cauth,
+							u_int i)
+{
+	echo_server_config_t *client, *server;
+
+	INIT(client,
+		.version = i,
+		.addr = "127.0.0.1",
+		.port = port,
+		.cauth = cauth,
+		.data = chunk_from_chars(0x01,0x02,0x03,0x04,0x05,0x06,0x07,0x08),
+	);
```

Comment:
Maybe a constructor to create such a config would make sense to avoid having to duplicate all these lines.